DownloaderForReddit/gui/add_reddit_object_dialog.py

```diff
@@ -0,0 +1,135 @@
+import os
+import logging
+from PyQt5.QtWidgets import QDialog, QFileDialog, QApplication
+from PyQt5.QtCore import Qt, pyqtSignal
+from pyqtspinner.spinner import WaitingSpinner
+from threading import Thread
+
+from ..guiresources.add_reddit_object_dialog_auto import Ui_AddRedditObjectDialog
+from ..utils import injector, system_util, reddit_utils
+from ..utils.importers import json_importer, text_importer
+
+
+class AddRedditObjectDialog(QDialog, Ui_AddRedditObjectDialog):
+
+    validation_finished = pyqtSignal()
+
+    def __init__(self, list_model, parent=None):
+        QDialog.__init__(self, parent=parent)
+        self.setupUi(self)
+        self.logger = logging.getLogger(f'DownloaderForReddit.{__name__}')
+        self.settings_manager = injector.get_settings_manager()
+        self.list_model = list_model
+        self.setWindowTitle(f'Add {self.list_model.list_type.capitalize()}')
+        self.single_add_label.setText(f'Enter new {self.list_model.list_type.lower()}')
+        self.tab_widget.setCurrentIndex(0)
+
+        self.download_on_add_checkbox.setChecked(self.settings_manager.download_on_add)
+        self.download_on_add_checkbox.stateChanged.connect(
+            lambda checked: setattr(self.settings_manager, 'download_on_add', checked))
+
+        self.add_button.clicked.connect(self.add_object_to_list)
+        self.remove_button.clicked.connect(self.remove_object_from_list)
+        self.import_button.clicked.connect(self.import_list)
+        self.dialog_button_box.accepted.connect(self.accept)
+        self.dialog_button_box.rejected.connect(self.close)
+
+        self.single_object_line_edit.setFocus()
+        self.tab_widget.currentChanged.connect(self.tab_change)
+
+        self.added = []
+        self.imported = []
+
+    def tab_change(self):
+        if self.tab_widget.currentIndex() == 0:
+            self.single_object_line_edit.setFocus()
+        else:
+            self.multi_object_line_edit.setFocus()
+
+    def refresh_name_count(self):
+        self.name_count_label.setText(str(self.multi_object_list_widget.count()))
+
+    def add_object_to_list(self):
+        name = self.multi_object_line_edit.text()
+        self.added.append(name)
+        self.multi_object_list_widget.addItem(name)
+        self.multi_object_line_edit.clear()
+        self.refresh_name_count()
+
+    def remove_object_from_list(self):
+        for index in self.multi_object_list_widget.selectedIndexes():
+            self.multi_object_list_widget.takeItem(index.row())
+        self.refresh_name_count()
+
+    def import_list(self):
+        file_path = self.get_import_file_path()
+        if file_path is not None:
```

Comment:
This crashes if the input file is empty, adding a check for empty files fixes this.
Something like 
`if os.stat(file_path).st_size != 0`

Reply:
Nice find, thanks!